src/cudadev/plugin-SiPixelRecHits/gpuPixelRecHits.h

```diff
@@ -30,14 +30,14 @@ namespace gpuPixelRecHits {
 
     auto& hits = *phits;
 
-    auto const digis = *pdigis;  // the copy is intentional!
-    auto const& clusters = *pclusters;
+    auto const digis = pdigis;  // the copy is intentional!
+    auto const& clusters = pclusters;
```

Comment:
Is either of these really needed anymore when the function gets the `pdigis` and `pclusters` by value?

Reply:
My understanding of line 33 is that we want to force a copy of the SoA structure from global memory to registers. Line 34 would require the same, were the `&` not be there. 

I tried without the copy/reference, original and with full copy (no `&` for the second SoA) and the kernel performances are indistinguishable (respectively):
```
                    4.28%  655.45ms     10000  65.545us  47.744us  446.01us  gpuPixelRecHits::getHits()
                    4.28%  652.34ms     10000  65.233us  46.912us  310.78us  gpuPixelRecHits::getHits()
                    4.30%  655.12ms     10000  65.512us  47.904us  374.08us  gpuPixelRecHits::getHits()
```
I removed the copies in 3986550.